---
title: "Identifying Biomarkers Impacting an Outcome Variable"
author: "Nima S. Hejazi & Alan E. Hubbard"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Identifying Biomarkers Impacting an Outcome Variable}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r core_reqs, echo=FALSE}
library(parallel)
library(doParallel)
library(foreach)
```

## Data Structure

```{r sim_data}
# specify key simulation parameters
set.seed(6423709)
n <- 100
g <- 2500

# generate outcome data
outcome <- rnorm(n = n, mean = 1000, sd = 50)

# generate expression measures related to outcome and those not
biomarkers_outcome <- replicate(n = n, rnorm(n = g/2, mean = sqrt(outcome), sd = 25))
biomarkers_unrelated <- replicate(n = n, rnorm(n = g/2, mean = 100, sd = 20))
biomarkers <- as.data.frame(rbind(biomarkers_outcome, biomarkers_unrelated))

# design matrix (of phenotype-level data) -- no relation to outcomes/biomarkers
design <- as.data.frame(replicate(4, rnorm(n = n, mean = 10, sd = 5)))
```

## Biomarker Identification

First, we load the `biotmle` package and some convenience dependencies:

```{r setup_data}
library(dplyr)
library(biotmle)
library(SummarizedExperiment)
```

In order to use Targeted Minimum Loss-Based Estimation to identify those
biomarkers related to the outcome, based on the Averate Treatment Effect (ATE)
parameter, we need three separate data structures: (1) _W_, baseline covariates
that could potentially confound the association of biomarkers with the outcome;
(2) _A_, a matrix of biomarkers, some associated with the outcome of interest
and others not; and (3) _Y_, the outcome of interest. All values in _W_ and _A_
ought to be discretized, in order to avoid practical violations of the
assumption of positivity. To illustrate, we discretize the phenotype-level data
below:

```{r discrete_W}
# discretize the phenotype-level data
W <- design %>%
  dplyr::mutate(V1 = as.numeric(V1 > median(V1)),
                V2 = as.numeric(V2 > quantile(V2)[2]),
                V3 = as.numeric(V3 < quantile(V3)[4]),
                V4 = as.numeric(V4 > quantile(V4)[1])) %>%
  DataFrame
```

To invoke the biomarker identification function (`biomarkertmle`), we also need
to discretize the biomarkers (again, this ensures that we do not violate the
assumption of positivity when assessing our target causal parameters, the ATE).

We'll use the following function to assign each of the expression measures in
the `data.frame` of biomarkers to the quartile of values, doing this over the
columns (subjects) of the `biomarkers` `data.frame`.

```{r discrete_by_quantile}
discrete_by_quantile <- function(x) {
  if( class(x) != "factor" ) {
    as.numeric(gtools::quantcut(x))
  } else {
    as.numeric(x)
  }
}
```

```{r discrete_A}
biomarkers_discrete <- as.data.frame(sapply(biomarkers, discrete_by_quantile))
```

```{r make_SE}
se <- SummarizedExperiment(assays = list(expMeasures = DataFrame(biomarkers_discrete)),
                           colData = DataFrame(cbind(outcome, design)))
se
```

The TMLE-based biomarker discovery process can be invoked using the
`biomarkertmle` function. The procedure is quite resource-intensive because it
evaluates the association of each individual potential biomarker with an
outcome of interest, while accounting for potential confounding based on all
other covariates included in the design matrix. We demonstrate the necessary
syntax for calling `biomarkertmle` below by applying to just a handful of
biomarkers:

```{r biomarkerTMLE_eval}
biomarkerTMLEout <- biomarkertmle(se = se[, seq_len(10)],
                                  varInt = 1,  # outcome: 1st "colData" variable
                                  type = "outcome",
                                  family = "gaussian",
                                  g_lib = c("SL.mean"),
                                  Q_lib = c("SL.mean")
                                 )
```

The output of `biomarkertmle` is an object of class `bioTMLE`, containing three
objects: (1) `call`, the call to `biomarkertmle`; (2) `topTable`, an empty
slot meant to hold the output of `limma::topTable`, after a later call to
`modtest_ic`; and (3) `tmleOut`, a data frame containing the point estimates of
the associations of each biomarker with the exposure of interest based on the
influence curve representation of the Average Treatment Effect parameter.

The output of `biomarkertmle` can be directly fed to `modtest_ic`, a wrapper
around `limma::lmFit` and `limma::topTable` that outputs a `biotmle` object
with the slots described above completely filled in. The `modtest_ic` function
requires as input a `biotmle` object containing a data frame in the `tmleOut`
field as well as a design matrix indicating the groupwise difference to be
tested. The design matrix should contain an intercept term and a term for the
exposure of interest (with discretized exposure levels). _Based on the relevant
statistical theory, it is not appropriate to include any further terms in the
design matrix (n.b., this differs from standard calls to `limma::lmFit`)_.

```{r limmaTMLE_eval}
limmaTMLEout <- modtest_ic(biotmle = biomarkerTMLEout)
```

After invoking `modtest_ic`, the resultant `bioTMLE` object will contain all
information relevant to the analytic procedure for identifying biomarkers: that
is, it will contain the origin call to `biomarkertmle`, the result of running
`limma::topTable`, the result of running `limma::lmFit`, and the result of
running `biomarkertmle`. The statistical results of this procedure can be
extracted from the `topTable` object in the `bioTMLE` object produced by
`modtest_ic`.

---

## Visualization of Results

This package provides several plotting methods that can be used to visualize
the results of the TMLE-based biomarker discovery process.

The `plot` method for a `bioTMLE` object will produce a histogram of the
adjusted p-values of each biomarker (based on the Benjamini-Hochberg procedure
for controlling the False Discovery Rate) as generated by `limma::topTable`:

```{r pval_hist_limma_adjp}
plot(x = limmaTMLEout, type = "pvals_adj")
```

Setting the argument `type = "pvals_raw"` will instead produce a histogram of
the raw p-values _(these are less informative and should, in general, not be
used for inferential purposes, as the computation producing these p-values
ignores the multiple testing nature of the biomarker discovery problem)_:

```{r pval_hist_limma_rawp}
plot(x = limmaTMLEout, type = "pvals_raw")
```

Heatmaps are useful graphics for visualizing the relationship between measures
on genomic objects and covariates of interest. The `heatmap_ic` function
provides this graphic for `bioTMLE` objects, allowing for the relationship
between the exposure variable and some number of "top" biomarkers (as
determined by the call to `modtest_ic`) to be visualized. In general, the
heatmap for `bioTMLE` objects expresses how the contributions of each biomarker
to the Average Treatment Effect (ATE) vary across differences in the exposure
variable (that is, there is a causal interpretation to the findings). The plot
produced is a `ggplot2` object and can be modified in place if stored properly.
For our analysis:

```{r heatmap_limma_results}
varInt_index <- which(names(colData(illuminaData)) %in% "benzene")
designVar <- as.data.frame(colData(illuminaData))[, varInt_index]
designVar <- as.numeric(designVar == max(designVar))

heatmap_ic(x = limmaTMLEout, design = designVar, FDRcutoff = 0.05, top = 25)
```

The volcano plot is standard graphical tools for examining how changes in
expression relate to the raw p-value. The utility of such plots lies in their
providing a convenient way to identify and systematically ignore those genomic
objects that have extremely low p-values due to extremely low variance between
observations. The `volcano_ic` function provides much of the same
interpretation, except that the fold change values displayed in the x-axis refer
to changes in the _contributions of each biomarker to the Average Treatment
Effect_ (in standard practice, for microarray technology, these would be fold
changes in gene expression). The plot produced is a `ggplot2` object and can
be modified in place if stored properly. For our analysis:

```{r volcano_plot_limma_results}
volcano_ic(biotmle = limmaTMLEout)
```

---

## Session Information

```{r sessionInfo, echo=FALSE}
sessionInfo()
```
