---
title: "Identifying Biomarkers from an Exposure Variable"
author: "Nima S. Hejazi & Alan E. Hubbard"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Identifying Biomarkers from an Exposure Variable}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

## Introduction

The `biotmle` R package can be used to isolate biomarkers in two ways: based
on the associations of genomic objects with (1) an exposure variable of interest
and (2) an outcome variable of interest. In this vignette, we illustrate how to
use `biotmle` to isolate and visualize genes associated with an __exposure__,
using a data set containing microarray expression measures from an Illumina
platform. In the analysis described below, Targeted Minimum Loss-Based
Estimation (TMLE) is used to transform the microarray expression values based
on the influence curve representation of the Average Treatment Effect (ATE).
Following this transformation, the moderated t-statistic of Smyth is used to
test for a binary groupwise difference (based on the exposure variable), using
the tools provided by the R package `limma`.

---

## Biomarker Identification

First, we load the `biotmle` package and the (included) `illuminaData` data set:

```{r setup_data}
library(dplyr)
library(biotmle)
data(illuminaData)
"%ni%" = Negate("%in%")
subjIDs <- illuminaData$id
```

In order to perform Targeted Minimum Loss-Based Estimation, we need three
separate data structures: (1) _W_, baseline covariates that could potentially
confound the association of biomarkers with the exposure of interest; (2) _A_,
the exposure of interest; and (3) _Y_, the biomarkers of interest. All values in
_W_ and _A_ must be discretized. We demonstrate the process of creating these
separate data structures with the `illuminaData` data set below:

```{r clean_data}
# W - age, sex, smoking
W <- illuminaData %>%
  dplyr::select(which(colnames(.) %in% c("age", "sex", "smoking"))) %>%
  dplyr::mutate(
    age = as.numeric((age > quantile(age, 0.25))),
    sex = I(sex),
    smoking = I(smoking)
  )


# A - benzene exposure (discretized)
A <- illuminaData %>%
  dplyr::select(which(colnames(.) %in% c("benzene")))
A <- A[, 1]


# Y - genes
Y <- illuminaData %>%
  dplyr::select(which(colnames(.) %ni% c("age", "sex", "smoking", "benzene",
                                         "id")))
geneIDs <- colnames(Y)
```

After creating _W_, _A_, and _Y_, the TMLE-based biomarker discovery process
can be invoked using the `biomarkertmle` function. The procedure is quite
resource-intensive because it evaluates the association of each individual
potential biomarker (of which there are over 20,000 in the included data set)
with exposure of interest _A_, while accounting for potential confounding based
on the covariates included in _W_. We demonstrate the necessary syntax for
calling `biomarkertmle` below:

```{r biomarkerTMLE_eval, eval=FALSE}
biomarkerTMLEout <- biomarkertmle(Y = Y,
                                  W = W,
                                  A = A,
                                  type = "exposure",
                                  parallel = TRUE,
                                  family = "gaussian",
                                  g_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.polymars", "SL.mean"),
                                  Q_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.nnet", "SL.mean")
                                 )
```

```{r load_biomarkerTMLE_result, echo=FALSE}
data(biomarkertmleOut)
```

The output of `biomarkertmle()` is an object of class `biotmle`, containing four
objects: (1) `call`, the call to `biomarkertmle()`; (2) `topTable`, an empty
slot meant to hold the output of `limma::topTable()`, after a later call to
`limmatmle()`; (3) `limmaOut`, an empty slot meant to hold the output of
`limma::lmFit()`, after a later call to `limmatmle()`; and (4) `tmleOut`, a data
frame containing the point estimates of the associations of each biomarker with
the exposure of interest based on the influence curve represenation of the
Average Treatment Effect parameter.

The output of `biomarkertmle()` can be directly fed to `limmatmle()`, a wrapper
around `limma::lmFit()` and `limma::topTable()` that outputs a `biotmle` object
with the slots described above completely filled in. The `limmatmle()` function
requires as input a `biotmle` object containing a data frame in the `tmleOut`
field as well as a design matrix indicating the groupwise difference to be
tested. The design matrix should contain an intercept term and a term for the
exposure of interest (with discretized exposure levels). _Based on the relevant
statistical theory, it is not appropriate to include any further terms in the
design matrix (n.b., this differs from standard calls to `limma::lmFit()`)_.

```{r limmaTMLE_eval}
design <- as.data.frame(cbind(rep(1, nrow(Y)),
                              as.numeric(A == max(unique(A)))))
colnames(design) <- c("intercept", "Tx")

limmaTMLEout <- limmatmle(biotmle = biomarkerTMLEout, IDs = NULL,
                          designMat = design)
```

After invoking `limmatmle()`, the resultant `biotmle` object will contain all
information relevant to the analytic procedure for identifying biomarkers: that
is, it will contain the origin call to `biomarkertmle()`, the result of running
`limma::topTable()`, the result of running `limma::lmFit()`, and the result of
running `biomarkertmle()`. The statistical results of this procedure can be
extracted from the `topTable` object in the `biotmle` object produced by
`limmatmle()`.

---

## Visualization of Results

This package provides several plotting methods that can be used to visualize
the results of the TMLE-based biomarker discovery process.

The `plot()` method for a `biotmle` object will produce a histogram of the
adjusted p-values of each biomarker (based on the Benjamini-Hochberg procedure
for controlling the False Discovery Rate) as generated by `limma::topTable()`:

```{r pval_hist_limma_adjp}
plot(x = limmaTMLEout, type = "pvals_adj")
```

Setting the argument `type="pvals_raw"` will instead produce a histogram of the
raw p-values _(these are less informative and should, in general, not be used
for inferential purposes, as the computation producing these p-values ignores
the multiple testing nature of the biomarker discovery problem)_:

```{r pval_hist_limma_rawp}
plot(x = limmaTMLEout, type = "pvals_raw")
```

Heatmaps are useful graphics for visualizing the relationship between measures
on genomic objects and covariates of interest. The `aheatmap.biotmle()` function
provides this graphic for `biotmle` objects, allowing for the relationship
between the exposure variable and some number of "top" biomarkers (as
determined by the call to `limmatmle()`) to be visualized. In general, the
heatmap for `biotmle` objects expresses how the contributions of each biomarker
to the Average Treatment Effect (ATE) vary across differences in the exposure
variable (that is, there is a causal interpretation to the findings). The plot
produced is a `ggplot2` object and can be modified in place if stored properly.
For our analysis:

```{r heatmap_limma_results}
aheatmap.biotmle(x = limmaTMLEout, designMat = design, FDRcutoff = 0.05,
                 top = 25)
```

The volcano plot is standard graphical tools for examining how changes in
expression relate to the raw p-value. The utility of such plots lies in their
providing a convenient way to identify and systematically ignore those genomic
objects that have extremely low p-values due to extremely low variance between
observations. The `volcanoPlot_biotmle()` function provides much of the same
interpretation, except that the fold change values displayed in the x-axis refer
to changes in the _contributions of each biomarker to the Average Treatment
Effect_ (in standard practice, for microarray technology, these would be fold
changes in gene expression). The plot produced is a `ggplot2` object and can
be modified in place if stored properly. For our analysis:

```{r volcano_plot_limma_results}
volcanoPlot_biotmle(biotmle = limmaTMLEout)
```
