<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Targeted Learning for Biomarker Discovery with Moderated Statistics • biotmle</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">biotmle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="/index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable</a>
    </li>
    <li>
      <a href="articles/rnaseqProcessing.html">Processing and Analyzing RNA-Seq Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nhejazi/biotmle">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    

    
    
<div class="contents">
<!-- README.md is generated from README.Rmd. Please edit that file -->
<section id="rbiotmle" class="level1"><div class="page-header"><h1>R/<code>biotmle</code>
</h1></div>

<blockquote>
<p>Targeted Learning with moderated statistics for biomarker discovery</p>
</blockquote>
<p><strong>Author:</strong> <a href="http://nimahejazi.org">Nima Hejazi</a></p>
<hr>
<section id="whats-biotmle" class="level2"><h2>What’s <code>biotmle</code>?</h2>
<p><code>biotmle</code> is an R package that facilitates biomarker discovery by generalizing the moderated t-statistic <span class="citation" data-cites="smyth2004linear">(Smyth 2004)</span> for use with target parameters that have asymptotically linear representations <span class="citation" data-cites="vdl2011targeted">(van der Laan and Rose 2011)</span>. The set of methods implemented in this R package rely on the use of Targeted Minimum Loss-Based Estimation (TMLE) to transform biological sequencing data (e.g., microarray, RNA-seq) based on the influence curve representation of a particular causal target parameter (e.g., Average Treatment Effect). The transformed data (rotated into influence curve space) may then be subjected to a moderated test for differences between the statistical estimate of the target parameter and a hypothesized value of said parameter (usually a null value defined in relation to the parameter itself). Such an approach provides a valid statistical hypothesis test of a statistically estimable causal parameter while controlling the variance such that the error rate (of the test) is more strongly controlled relative to testing procedures that do not moderate the variance estimate <span class="citation" data-cites="hejazi2018variance">(Hejazi et al., n.d.)</span>.</p>
<hr></section><section id="installation" class="level2"><h2>Installation</h2>
<p>For standard use, install from <a href="https://bioconductor.org">Bioconductor</a>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">biocLite</span>(<span class="st">"biotmle"</span>)</a></code></pre></div>
<p>To contribute, install the bleeding-edge <em>development version</em> from GitHub via <a href="https://www.rstudio.com/products/rpackages/devtools/"><code>devtools</code></a>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"nhejazi/biotmle"</span>)</a></code></pre></div>
<p>Current and prior <a href="https://bioconductor.org">Bioconductor</a> releases are available under branches with numbers prefixed by “RELEASE_”. For example, to install the version of this package available via Bioconductor 3.6, use</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"nhejazi/biotmle"</span>, <span class="dt">ref =</span> <span class="st">"RELEASE_3_6"</span>)</a></code></pre></div>
<hr></section><section id="issues" class="level2"><h2>Issues</h2>
<p>If you encounter any bugs or have any specific feature requests, please <a href="https://github.com/nhejazi/biotmle/issues">file an issue</a>.</p>
<hr></section><section id="contributions" class="level2"><h2>Contributions</h2>
<p>It is our hope that <code>biotmle</code> will grow to be widely adopted as a tool for the nonparametric assessment of variable importance in a variety of settings involving data generated by experiments in computational biology. To that end, contributions are very welcome, though we ask that interested contributors consult our <a href="https://github.com/nhejazi/biotmle/blob/master/CONTRIBUTING.md"><code>contribution guidelines</code></a> prior to submitting a pull request.</p>
<hr></section><section id="citation" class="level2"><h2>Citation</h2>
<p>After using the <code>biotmle</code> R package, please cite it:</p>
<pre><code>    @article{hejazi2017biotmle,
      author = {Hejazi, Nima S and Cai, Weixin and Hubbard, Alan E},
      title = {biotmle: Targeted Learning for Biomarker Discovery},
      journal = {The Journal of Open Source Software},
      volume = {2},
      number = {15},
      month = {July},
      year  = {2017},
      publisher = {The Open Journal},
      doi = {10.21105/joss.00295},
      url = {https://doi.org/10.21105/joss.00295}
    }</code></pre>
<hr></section><section id="related" class="level2"><h2>Related</h2>
<ul>
<li>
<a href="https://github.com/nhejazi/biotmleData">R/<code>biotmleData</code></a> - R package with example experimental data for use with this analysis package.</li>
</ul>
<hr></section><section id="funding" class="level2"><h2>Funding</h2>
<p>The development of this software was supported in part through a grant from the National Institutes of Health: <a href="https://projectreporter.nih.gov/project_info_details.cfm?aid=9260357&amp;map=y">P42 ES004705-29</a>.</p>
<hr></section><section id="license" class="level2"><h2>License</h2>
<p>© 2016-2018 <a href="https://nimahejazi.org">Nima S. Hejazi</a> &amp; <a href="http://hubbard.berkeley.edu/">Alan E. Hubbard</a></p>
<p>The contents of this repository are distributed under the MIT license. See file <code>LICENSE</code> for details.</p>
<hr></section><section id="references" class="level2 unnumbered"><h2>References</h2>
<div id="refs" class="references">
<div id="ref-hejazi2018variance">
<p>Hejazi, Nima S, Sara Kherad-Pajouh, Mark J van der Laan, and Alan E Hubbard. n.d. “Variance Stabilization of Targeted Estimators of Causal Parameters in High-Dimensional Settings.” <a href="https://arxiv.org/abs/1710.05451" class="uri">https://arxiv.org/abs/1710.05451</a>.</p>
</div>
<div id="ref-smyth2004linear">
<p>Smyth, Gordon K. 2004. “Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments.” <em>Statistical Applications in Genetics and Molecular Biology</em> 3 (1). Walter de Gruyter:1–25. <a href="https://doi.org/10.2202/1544-6115.1027" class="uri">https://doi.org/10.2202/1544-6115.1027</a>.</p>
</div>
<div id="ref-vdl2011targeted">
<p>van der Laan, Mark J., and Sherri Rose. 2011. <em>Targeted Learning: Causal Inference for Observational and Experimental Data</em>. Springer Science &amp; Business Media.</p>
</div>
</div>
</section></section>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2 class="hasAnchor">
<a href="#sidebar" class="anchor"></a>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/biotmle">https://​www.bioconductor.org/​packages/​biotmle</a>
</li>
<li>Browse source code at <br><a href="https://github.com/nhejazi/biotmle">https://​github.com/​nhejazi/​biotmle</a>
</li>
<li>Report a bug at <br><a href="https://github.com/nhejazi/biotmle/issues">https://​github.com/​nhejazi/​biotmle/​issues</a>
</li>
</ul>
<h2>License</h2>
<p>file <a href="LICENSE.html">LICENSE</a></p>
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing biotmle</a></li>
</ul>
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Nima Hejazi <br><small class="roles"> Author, maintainer, copyright holder </small> <br><small>(0000-0002-7127-2789)</small>
</li>
<li>Alan Hubbard <br><small class="roles"> Author </small> <br><small>(0000-0002-3769-0127)</small>
</li>
<li><a href="authors.html">All authors...</a></li>
</ul>
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/nhejazi/biotmle"><img src="https://travis-ci.org/nhejazi/biotmle.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://ci.appveyor.com/project/nhejazi/biotmle/"><img src="https://ci.appveyor.com/api/projects/status/github/nhejazi/biotmle?branch=master&amp;svg=true" alt="AppVeyor Build Status"></a></li>
<li><a href="https://codecov.io/github/nhejazi/biotmle?branch=master"><img src="https://img.shields.io/codecov/c/github/nhejazi/biotmle/master.svg" alt="Coverage Status"></a></li>
<li><a href="http://www.repostatus.org/#active"><img src="http://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/biotmle.html"><img src="http://bioconductor.org/shields/years-in-bioc/biotmle.svg" alt="Bioc Time"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/biotmle.html"><img src="http://bioconductor.org/shields/downloads/biotmle.svg" alt="Bioc Downloads"></a></li>
<li><a href="http://opensource.org/licenses/MIT"><img src="http://img.shields.io/badge/license-MIT-brightgreen.svg" alt="MIT license"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/65854775"><img src="https://zenodo.org/badge/65854775.svg" alt="DOI"></a></li>
<li><a href="http://joss.theoj.org/papers/02be843d9bab1b598187bfbb08ce3949"><img src="http://joss.theoj.org/papers/02be843d9bab1b598187bfbb08ce3949/status.svg" alt="JOSS Status"></a></li>
</ul>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
