<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Identifying Biomarkers from an Exposure Variable • biotmle</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">biotmle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable</a>
    </li>
    <li>
      <a href="../articles/rnaseqProcessing.html">Processing and Analyzing RNA-Seq Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nhejazi/biotmle">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Identifying Biomarkers from an Exposure Variable</h1>
                        <h4 class="author">Nima Hejazi &amp; Alan Hubbard</h4>
            
            <h4 class="date">2017-07-28</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The <code>biotmle</code> R package can be used to isolate biomarkers in two ways: based on the associations of genomic objects with an exposure variable of interest. In this vignette, we illustrate how to use <code>biotmle</code> to isolate and visualize genes associated with an <strong>exposure</strong>, using a data set containing microarray expression measures from an Illumina platform. In the analysis described below, Targeted Minimum Loss-Based Estimation (TMLE) is used to transform the microarray expression values based on the influence curve representation of the Average Treatment Effect (ATE). Following this transformation, the moderated t-statistic of Smyth is used to test for a binary groupwise difference (based on the exposure variable), using the tools provided by the R package <code>limma</code>.</p>
<hr>
</div>
<div id="biomarker-identification" class="section level2">
<h2 class="hasAnchor">
<a href="#biomarker-identification" class="anchor"></a>Biomarker Identification</h2>
<p>First, we load the <code>biotmle</code> package and the (included) <code>illuminaData</code> data set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dplyr)</code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(biotmle)</code></pre></div>
<pre><code>## biotmle: Moderated Statistics and Targeted Learning for Biomarker Discovery</code></pre>
<pre><code>## Version: 1.1.2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(biotmleData)
<span class="kw">data</span>(illuminaData)
<span class="kw">library</span>(SummarizedExperiment)</code></pre></div>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: 'IRanges'</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     collapse, desc, slice</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## The following object is masked from 'package:dplyr':
## 
##     count</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     apply</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="st">"%ni%"</span> =<span class="st"> </span><span class="kw">Negate</span>(<span class="st">"%in%"</span>)</code></pre></div>
<p>In order to perform Targeted Minimum Loss-Based Estimation, we need three separate data structures: (1) <em>W</em>, baseline covariates that could potentially confound the association of biomarkers with the exposure of interest; (2) <em>A</em>, the exposure of interest; and (3) <em>Y</em>, the biomarkers of interest. All values in <em>W</em> and <em>A</em> ought to be discretized, in order to avoid practical violations of the assumption of positivity. With the <code>illuminaData</code> data set below, we discretize the age variable in the phenotype-level data below (this can be accessed via the <code>colData</code> of the <code>SummarizedExperiment</code> object). To invoke the biomarker assessment function (<code>biomarkertmle</code>), we also need to specify a variable of interest (or the position of said variable in the design matrix). We do both in just a few lines below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># discretize "age" in the phenotype-level data</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(illuminaData) &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(illuminaData) <span class="op">%&gt;%</span>
<span class="st">  </span>data.frame <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">age =</span> <span class="kw">as.numeric</span>(age <span class="op">&gt;</span><span class="st"> </span><span class="kw">median</span>(age))) <span class="op">%&gt;%</span>
<span class="st">  </span>DataFrame

<span class="co"># specify column index of treatment/exposure variable of interest</span>
varInt_index &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">names</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(illuminaData)) <span class="op">%in%</span><span class="st"> "benzene"</span>)</code></pre></div>
<p>The TMLE-based biomarker discovery process can be invoked using the <code>biomarkertmle</code> function. The procedure is quite resource-intensive because it evaluates the association of each individual potential biomarker (of which there are over 20,000 in the included data set) with an exposure of interest, while accounting for potential confounding based on all other covariates included in the design matrix. We demonstrate the necessary syntax for calling <code>biomarkertmle</code> below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">biomarkerTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/biomarkertmle.html">biomarkertmle</a></span>(<span class="dt">se =</span> illuminaData,
                                  <span class="dt">varInt =</span> varInt_index,
                                  <span class="dt">family =</span> <span class="st">"gaussian"</span>,
                                  <span class="dt">g_lib =</span> <span class="kw">c</span>(<span class="st">"SL.glmnet"</span>, <span class="st">"SL.randomForest"</span>,
                                            <span class="st">"SL.polymars"</span>, <span class="st">"SL.mean"</span>),
                                  <span class="dt">Q_lib =</span> <span class="kw">c</span>(<span class="st">"SL.glmnet"</span>, <span class="st">"SL.randomForest"</span>,
                                            <span class="st">"SL.nnet"</span>, <span class="st">"SL.mean"</span>)
                                 )</code></pre></div>
<p>The output of <code>biomarkertmle</code> is an object of class <code>bioTMLE</code>, containing four objects: (1) <code>call</code>, the call to <code>biomarkertmle</code>; (2) <code>topTable</code>, an empty slot meant to hold the output of <code><a href="http://www.rdocumentation.org/packages/limma/topics/toptable">limma::topTable</a></code>, after a later call to <code>modtest_ic</code>; (3) <code>limmaOut</code>, an empty slot meant to hold the output of <code><a href="http://www.rdocumentation.org/packages/limma/topics/lmFit">limma::lmFit</a></code>, after a later call to <code>modtest_ic</code>; and (4) <code>tmleOut</code>, a data frame containing the point estimates of the associations of each biomarker with the exposure of interest based on the influence curve representation of the Average Treatment Effect parameter.</p>
<p>The output of <code>biomarkertmle</code> can be directly fed to <code>modtest_ic</code>, a wrapper around <code><a href="http://www.rdocumentation.org/packages/limma/topics/lmFit">limma::lmFit</a></code> and <code><a href="http://www.rdocumentation.org/packages/limma/topics/toptable">limma::topTable</a></code> that outputs a <code>biotmle</code> object with the slots described above completely filled in. The <code>modtest_ic</code> function requires as input a <code>biotmle</code> object containing a data frame in the <code>tmleOut</code> field as well as a design matrix indicating the groupwise difference to be tested. The design matrix should contain an intercept term and a term for the exposure of interest (with discretized exposure levels). <em>Based on the relevant statistical theory, it is not appropriate to include any further terms in the design matrix (n.b., this differs from standard calls to <code><a href="http://www.rdocumentation.org/packages/limma/topics/lmFit">limma::lmFit</a></code>)</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">limmaTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/modtest_ic.html">modtest_ic</a></span>(<span class="dt">biotmle =</span> biomarkerTMLEout)</code></pre></div>
<p>After invoking <code>modtest_ic</code>, the resultant <code>bioTMLE</code> object will contain all information relevant to the analytic procedure for identifying biomarkers: that is, it will contain the origin call to <code>biomarkertmle</code>, the result of running <code><a href="http://www.rdocumentation.org/packages/limma/topics/toptable">limma::topTable</a></code>, the result of running <code><a href="http://www.rdocumentation.org/packages/limma/topics/lmFit">limma::lmFit</a></code>, and the result of running <code>biomarkertmle</code>. The statistical results of this procedure can be extracted from the <code>topTable</code> object in the <code>bioTMLE</code> object produced by <code>modtest_ic</code>.</p>
<hr>
</div>
<div id="visualization-of-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization-of-results" class="anchor"></a>Visualization of Results</h2>
<p>This package provides several plotting methods that can be used to visualize the results of the TMLE-based biomarker discovery process.</p>
<p>The <code>plot</code> method for a <code>bioTMLE</code> object will produce a histogram of the adjusted p-values of each biomarker (based on the Benjamini-Hochberg procedure for controlling the False Discovery Rate) as generated by <code><a href="http://www.rdocumentation.org/packages/limma/topics/toptable">limma::topTable</a></code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_adj"</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/pval_hist_limma_adjp-1.png" width="672"></p>
<p>Setting the argument <code>type = "pvals_raw"</code> will instead produce a histogram of the raw p-values <em>(these are less informative and should, in general, not be used for inferential purposes, as the computation producing these p-values ignores the multiple testing nature of the biomarker discovery problem)</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_raw"</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/pval_hist_limma_rawp-1.png" width="672"></p>
<p>Heatmaps are useful graphics for visualizing the relationship between measures on genomic objects and covariates of interest. The <code>heatmap_ic</code> function provides this graphic for <code>bioTMLE</code> objects, allowing for the relationship between the exposure variable and some number of “top” biomarkers (as determined by the call to <code>modtest_ic</code>) to be visualized. In general, the heatmap for <code>bioTMLE</code> objects expresses how the contributions of each biomarker to the Average Treatment Effect (ATE) vary across differences in the exposure variable (that is, there is a causal interpretation to the findings). The plot produced is a <code>ggplot2</code> object and can be modified in place if stored properly. For our analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varInt_index &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">names</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(illuminaData)) <span class="op">%in%</span><span class="st"> "benzene"</span>)
designVar &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(illuminaData))[, varInt_index]
designVar &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(designVar <span class="op">==</span><span class="st"> </span><span class="kw">max</span>(designVar))

<span class="kw"><a href="../reference/heatmap_ic.html">heatmap_ic</a></span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">design =</span> designVar, <span class="dt">FDRcutoff =</span> <span class="fl">0.05</span>, <span class="dt">top =</span> <span class="dv">25</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/heatmap_limma_results-1.png" width="672"></p>
<p>The volcano plot is standard graphical tools for examining how changes in expression relate to the raw p-value. The utility of such plots lies in their providing a convenient way to identify and systematically ignore those genomic objects that have extremely low p-values due to extremely low variance between observations. The <code>volcano_ic</code> function provides much of the same interpretation, except that the fold change values displayed in the x-axis refer to changes in the <em>contributions of each biomarker to the Average Treatment Effect</em> (in standard practice, for microarray technology, these would be fold changes in gene expression). The plot produced is a <code>ggplot2</code> object and can be modified in place if stored properly. For our analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/volcano_ic.html">volcano_ic</a></span>(<span class="dt">biotmle =</span> limmaTMLEout)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/volcano_plot_limma_results-1.png" width="672"></p>
<hr>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<pre><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-apple-darwin16.6.0 (64-bit)
## Running under: macOS Sierra 10.12.5
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2               SummarizedExperiment_1.6.3
##  [3] DelayedArray_0.2.7         matrixStats_0.52.2        
##  [5] Biobase_2.36.2             GenomicRanges_1.28.4      
##  [7] GenomeInfoDb_1.12.2        IRanges_2.10.2            
##  [9] S4Vectors_0.14.3           BiocGenerics_0.22.0       
## [11] biotmleData_0.99.1         biotmle_1.1.2             
## [13] dplyr_0.7.2               
## 
## loaded via a namespace (and not attached):
##  [1] superheat_0.1.0          lattice_0.20-35         
##  [3] colorspace_1.3-2         htmltools_0.3.6         
##  [5] yaml_2.1.14              rlang_0.1.1.9000        
##  [7] glue_1.1.1               DBI_0.7                 
##  [9] GenomeInfoDbData_0.99.0  foreach_1.4.3           
## [11] bindr_0.1                plyr_1.8.4              
## [13] stringr_1.2.0.9000       zlibbioc_1.22.0         
## [15] munsell_0.4.3            gtable_0.2.0            
## [17] tmle_1.2.0-5             codetools_0.2-15        
## [19] evaluate_0.10.1          labeling_0.3            
## [21] knitr_1.16               doParallel_1.0.10       
## [23] Rcpp_0.12.12             backports_1.1.0         
## [25] scales_0.4.1.9002        limma_3.32.4            
## [27] XVector_0.16.0           ggplot2_2.2.1           
## [29] digest_0.6.12            stringi_1.1.5           
## [31] grid_3.4.1               rprojroot_1.2           
## [33] tools_3.4.1              bitops_1.0-6            
## [35] magrittr_1.5             RCurl_1.95-4.8          
## [37] lazyeval_0.2.0           tibble_1.3.3.9001       
## [39] ggdendro_0.1-20          wesanderson_0.3.4       
## [41] pkgconfig_2.0.1          MASS_7.3-47             
## [43] Matrix_1.2-10            data.table_1.10.4       
## [45] SuperLearner_2.0-23-9000 nnls_1.4                
## [47] assertthat_0.2.0         rmarkdown_1.6           
## [49] iterators_1.0.8          R6_2.2.2                
## [51] compiler_3.4.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#biomarker-identification">Biomarker Identification</a></li>
      <li><a href="#visualization-of-results">Visualization of Results</a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
