<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Identifying Biomarkers from an Exposure Variable • biotmle</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">biotmle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nhejazi/biotmle">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Identifying Biomarkers from an Exposure Variable</h1>
                        <h4 class="author">Nima S. Hejazi &amp; Alan E. Hubbard</h4>
            
            <h4 class="date">2017-01-03</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#introduction" class="anchor"> </a></body></html>Introduction</h2>
<p>The <code>biotmle</code> R package can be used to isolate biomarkers in two ways: based on the associations of genomic objects with (1) an exposure variable of interest and (2) an outcome variable of interest. In this vignette, we illustrate how to use <code>biotmle</code> to isolate and visualize genes associated with an <strong>exposure</strong>, using a data set containing microarray expression measures from an Illumina platform. In the analysis described below, Targeted Minimum Loss-Based Estimation (TMLE) is used to transform the microarray expression values based on the influence curve representation of the Average Treatment Effect (ATE). Following this transformation, the moderated t-statistic of Smyth is used to test for a binary groupwise difference (based on the exposure variable), using the tools provided by the R package <code>limma</code>.</p>
<hr>
</div>
<div id="biomarker-identification" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#biomarker-identification" class="anchor"> </a></body></html>Biomarker Identification</h2>
<p>First, we load the <code>biotmle</code> package and the (included) <code>illuminaData</code> data set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dplyr)
<span class="kw">library</span>(biotmle)
<span class="kw">data</span>(illuminaData)
<span class="st">"%ni%"</span> =<span class="st"> </span><span class="kw">Negate</span>(<span class="st">"%in%"</span>)
subjIDs &lt;-<span class="st"> </span>illuminaData$id</code></pre></div>
<p>In order to perform Targeted Minimum Loss-Based Estimation, we need three separate data structures: (1) <em>W</em>, baseline covariates that could potentially confound the association of biomarkers with the exposure of interest; (2) <em>A</em>, the exposure of interest; and (3) <em>Y</em>, the biomarkers of interest. All values in <em>W</em> and <em>A</em> must be discretized. We demonstrate the process of creating these separate data structures with the <code>illuminaData</code> data set below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># W - age, sex, smoking</span>
W &lt;-<span class="st"> </span>illuminaData %&gt;%
<span class="st">  </span>dplyr::<span class="kw">select</span>(<span class="kw">which</span>(<span class="kw">colnames</span>(.) %in%<span class="st"> </span><span class="kw">c</span>(<span class="st">"age"</span>, <span class="st">"sex"</span>, <span class="st">"smoking"</span>))) %&gt;%
<span class="st">  </span>dplyr::<span class="kw">mutate</span>(
    <span class="dt">age =</span> <span class="kw">as.numeric</span>((age &gt;<span class="st"> </span><span class="kw">quantile</span>(age, <span class="fl">0.25</span>))),
    <span class="dt">sex =</span> <span class="kw">I</span>(sex),
    <span class="dt">smoking =</span> <span class="kw">I</span>(smoking)
  )


<span class="co"># A - benzene exposure (discretized)</span>
A &lt;-<span class="st"> </span>illuminaData %&gt;%
<span class="st">  </span>dplyr::<span class="kw">select</span>(<span class="kw">which</span>(<span class="kw">colnames</span>(.) %in%<span class="st"> </span><span class="kw">c</span>(<span class="st">"benzene"</span>)))
A &lt;-<span class="st"> </span>A[, <span class="dv">1</span>]


<span class="co"># Y - genes</span>
Y &lt;-<span class="st"> </span>illuminaData %&gt;%
<span class="st">  </span>dplyr::<span class="kw">select</span>(<span class="kw">which</span>(<span class="kw">colnames</span>(.) %ni%<span class="st"> </span><span class="kw">c</span>(<span class="st">"age"</span>, <span class="st">"sex"</span>, <span class="st">"smoking"</span>, <span class="st">"benzene"</span>,
                                         <span class="st">"id"</span>)))
geneIDs &lt;-<span class="st"> </span><span class="kw">colnames</span>(Y)</code></pre></div>
<p>After creating <em>W</em>, <em>A</em>, and <em>Y</em>, the TMLE-based biomarker discovery process can be invoked using the <code>biomarkertmle</code> function. The procedure is quite resource-intensive because it evaluates the association of each individual potential biomarker (of which there are over 20,000 in the included data set) with exposure of interest <em>A</em>, while accounting for potential confounding based on the covariates included in <em>W</em>. We demonstrate the necessary syntax for calling <code>biomarkertmle</code> below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">biomarkerTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/biomarkertmle.html">biomarkertmle</a></span>(<span class="dt">Y =</span> Y,
                                  <span class="dt">W =</span> W,
                                  <span class="dt">A =</span> A,
                                  <span class="dt">type =</span> <span class="st">"exposure"</span>,
                                  <span class="dt">parallel =</span> <span class="ot">TRUE</span>,
                                  <span class="dt">family =</span> <span class="st">"gaussian"</span>,
                                  <span class="dt">g_lib =</span> <span class="kw">c</span>(<span class="st">"SL.glmnet"</span>, <span class="st">"SL.randomForest"</span>,
                                            <span class="st">"SL.polymars"</span>, <span class="st">"SL.mean"</span>),
                                  <span class="dt">Q_lib =</span> <span class="kw">c</span>(<span class="st">"SL.glmnet"</span>, <span class="st">"SL.randomForest"</span>,
                                            <span class="st">"SL.nnet"</span>, <span class="st">"SL.mean"</span>)
                                 )</code></pre></div>
<p>The output of <code><a href="../reference/biomarkertmle.html">biomarkertmle()</a></code> is an object of class <code>biotmle</code>, containing four objects: (1) <code>call</code>, the call to <code><a href="../reference/biomarkertmle.html">biomarkertmle()</a></code>; (2) <code>topTable</code>, an empty slot meant to hold the output of <code>limma::topTable()</code>, after a later call to <code><a href="../reference/limmatmle.html">limmatmle()</a></code>; (3) <code>limmaOut</code>, an empty slot meant to hold the output of <code>limma::lmFit()</code>, after a later call to <code><a href="../reference/limmatmle.html">limmatmle()</a></code>; and (4) <code>tmleOut</code>, a data frame containing the point estimates of the associations of each biomarker with the exposure of interest based on the influence curve represenation of the Average Treatment Effect parameter.</p>
<p>The output of <code><a href="../reference/biomarkertmle.html">biomarkertmle()</a></code> can be directly fed to <code><a href="../reference/limmatmle.html">limmatmle()</a></code>, a wrapper around <code>limma::lmFit()</code> and <code>limma::topTable()</code> that outputs a <code>biotmle</code> object with the slots described above completely filled in. The <code><a href="../reference/limmatmle.html">limmatmle()</a></code> function requires as input a <code>biotmle</code> object containing a data frame in the <code>tmleOut</code> field as well as a design matrix indicating the groupwise difference to be tested. The design matrix should contain an intercept term and a term for the exposure of interest (with discretized exposure levels). <em>Based on the relevant statistical theory, it is not appropriate to include any further terms in the design matrix (n.b., this differs from standard calls to <code>limma::lmFit()</code>)</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(Y)),
                              <span class="kw">as.numeric</span>(A ==<span class="st"> </span><span class="kw">max</span>(<span class="kw">unique</span>(A)))))
<span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"intercept"</span>, <span class="st">"Tx"</span>)

limmaTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/limmatmle.html">limmatmle</a></span>(<span class="dt">biotmle =</span> biomarkerTMLEout, <span class="dt">IDs =</span> <span class="ot">NULL</span>,
                          <span class="dt">designMat =</span> design)</code></pre></div>
<p>After invoking <code><a href="../reference/limmatmle.html">limmatmle()</a></code>, the resultant <code>biotmle</code> object will contain all information relevant to the analytic procedure for identifying biomarkers: that is, it will contain the origin call to <code><a href="../reference/biomarkertmle.html">biomarkertmle()</a></code>, the result of running <code>limma::topTable()</code>, the result of running <code>limma::lmFit()</code>, and the result of running <code><a href="../reference/biomarkertmle.html">biomarkertmle()</a></code>. The statistical results of this procedure can be extracted from the <code>topTable</code> object in the <code>biotmle</code> object produced by <code><a href="../reference/limmatmle.html">limmatmle()</a></code>.</p>
<hr>
</div>
<div id="visualization-of-results" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#visualization-of-results" class="anchor"> </a></body></html>Visualization of Results</h2>
<p>This package provides several plotting methods that can be used to visualize the results of the TMLE-based biomarker discovery process.</p>
<p>The <code>plot()</code> method for a <code>biotmle</code> object will produce a histogram of the adjusted p-values of each biomarker (based on the Benjamini-Hochberg procedure for controlling the False Discovery Rate) as generated by <code>limma::topTable()</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_adj"</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/pval_hist_limma_adjp-1.png" width="672"></p>
<p>Setting the argument <code>type="pvals_raw"</code> will instead produce a histogram of the raw p-values <em>(these are less informative and should, in general, not be used for inferential purposes, as the computation producing these p-values ignores the multiple testing nature of the biomarker discovery problem)</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_raw"</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/pval_hist_limma_rawp-1.png" width="672"></p>
<p>Heatmaps are useful graphics for visualizing the relationship between measures on genomic objects and covariates of interest. The <code><a href="../reference/aheatmap.biotmle.html">aheatmap.biotmle()</a></code> function provides this graphic for <code>biotmle</code> objects, allowing for the relationship between the exposure variable and some number of “top” biomarkers (as determined by the call to <code><a href="../reference/limmatmle.html">limmatmle()</a></code>) to be visualized. In general, the heatmap for <code>biotmle</code> objects expresses how the contributions of each biomarker to the Average Treatment Effect (ATE) vary across differences in the exposure variable (that is, there is a causal interpretation to the findings). The plot produced is a <code>ggplot2</code> object and can be modified in place if stored properly. For our analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/aheatmap.biotmle.html">aheatmap.biotmle</a></span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">designMat =</span> design, <span class="dt">FDRcutoff =</span> <span class="fl">0.05</span>,
                 <span class="dt">top =</span> <span class="dv">25</span>)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/heatmap_limma_results-1.png" width="672"></p>
<p>The volcano plot is standard graphical tools for examining how changes in expression relate to the raw p-value. The utility of such plots lies in their providing a convenient way to identify and systematically ignore those genomic objects that have extremely low p-values due to extremely low variance between observations. The <code><a href="../reference/volcanoPlot_biotmle.html">volcanoPlot_biotmle()</a></code> function provides much of the same interpretation, except that the fold change values displayed in the x-axis refer to changes in the <em>contributions of each biomarker to the Average Treatment Effect</em> (in standard practice, for microarray technology, these would be fold changes in gene expression). The plot produced is a <code>ggplot2</code> object and can be modified in place if stored properly. For our analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/volcanoPlot_biotmle.html">volcanoPlot_biotmle</a></span>(<span class="dt">biotmle =</span> limmaTMLEout)</code></pre></div>
<p><img src="exposureBiomarkers_files/figure-html/volcano_plot_limma_results-1.png" width="672"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#biomarker-identification">Biomarker Identification</a></li>
      <li><a href="#visualization-of-results">Visualization of Results</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nima S. Hejazi, Alan E. Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
