<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing and Analyzing RNA-Seq Data â€¢ biotmle</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing and Analyzing RNA-Seq Data">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biotmle</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable</a>
    </li>
    <li>
      <a href="../articles/rnaseqProcessing.html">Processing and Analyzing RNA-Seq Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Processing and Analyzing RNA-Seq Data</h1>
                        <h4 class="author">Nima Hejazi</h4>
            
            <h4 class="date">2019-04-03</h4>
      
      
      <div class="hidden name"><code>rnaseqProcessing.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(biotmle)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(biotmleData)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SummarizedExperiment))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="st">"%ni%"</span> =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/funprog">Negate</a></span>(<span class="st">"%in%"</span>)</a></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Here, we briefly work through how to use the <code>biotmle</code> package with data generated by next-generation sequencing technologies, which, in contrast to microarray technologies, produce measurements in the form of discrete counts.</p>
<hr>
</div>
<div id="simulation-data-structure" class="section level2">
<h2 class="hasAnchor">
<a href="#simulation-data-structure" class="anchor"></a>Simulation: Data Structure</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">6423709</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">n &lt;-<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">g &lt;-<span class="st"> </span><span class="dv">2500</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">cases_pois &lt;-<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5">controls_pois &lt;-<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">ngs_cases &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">replicate</a></span>(n, <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Poisson">rpois</a></span>(g, cases_pois)), g))</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">ngs_controls &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">replicate</a></span>(n, <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Poisson">rpois</a></span>(g, controls_pois)), g))</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">ngs_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">cbind</a></span>(ngs_cases, ngs_controls))</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">exp_var &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">1</span>, n), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">0</span>, n))</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">batch &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, n)</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">covar &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">1</span>, n <span class="op">*</span><span class="st"> </span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb2-14" data-line-number="14">design &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">cbind</a></span>(exp_var, batch, covar))</a>
<a class="sourceLine" id="cb2-15" data-line-number="15"></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(ngs_data[, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>])</a></code></pre></div>
<pre><code>##   V1 V2 V3 V4 V5 V6 V7
## 1 69 61 39 46 50 57 43
## 2 42 59 50 54 44 53 54
## 3 41 49 52 54 50 58 34
## 4 30 44 49 44 46 36 61
## 5 50 56 45 44 46 61 58
## 6 61 50 47 53 49 54 67</code></pre>
<hr>
</div>
<div id="processing-data-transformation" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-data-transformation" class="anchor"></a>Processing: Data Transformation</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">se &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/RangedSummarizedExperiment-class">SummarizedExperiment</a></span>(<span class="dt">assays =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">counts =</span> <span class="kw">DataFrame</span>(ngs_data)),</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                           <span class="dt">colData =</span> <span class="kw">DataFrame</span>(design))</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">se</a></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 2500 100 
## metadata(0):
## assays(1): counts
## rownames: NULL
## rowData names(0):
## colnames(100): V1 V2 ... V49.1 V50.1
## colData names(3): exp_var batch covar</code></pre>
<hr>
</div>
<div id="analysis-assessing-the-effect-of-exposure" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-assessing-the-effect-of-exposure" class="anchor"></a>Analysis: Assessing the Effect of Exposure</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">rnaseqTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/biomarkertmle.html">biomarkertmle</a></span>(<span class="dt">se =</span> se,</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                               <span class="dt">varInt =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                               <span class="dt">ngscounts =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                               <span class="dt">parallel =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">                               <span class="dt">family =</span> <span class="st">"gaussian"</span>,</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">                               <span class="dt">g_lib =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"SL.mean"</span>, <span class="st">"SL.glm"</span>,</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">                                         <span class="st">"SL.randomForest"</span>),</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">                               <span class="dt">Q_lib =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"SL.mean"</span>, <span class="st">"SL.glm"</span>,</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">                                         <span class="st">"SL.randomForest"</span>, <span class="st">"SL.nnet"</span>)</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">                              )</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="../reference/eif.html">eif</a></span>(rnaseqTMLEout)<span class="op">$</span>E[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_len</a></span>(<span class="dv">6</span>)])</a></code></pre></div>
<pre><code>##                [,1]       [,2]      [,3]       [,4]      [,5]       [,6]
## result.1 -329.83655 -212.28070 160.39144   32.96832 -36.05786 -146.77124
## result.2   91.43805 -129.00196 -39.20680  -47.18052  44.29473  -30.83862
## result.3  102.01072 -294.65959 -77.65629 -376.36991 -61.66602 -441.74551
## result.4  323.94443   93.76761  13.80324   93.71017  47.56326  224.33603
## result.5   11.71456 -148.33216  93.43461   47.79556  61.82086 -229.99698
## result.6 -183.73280   -4.00971  45.02578  -53.02987  -4.00971  -69.37162</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">limmaTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/modtest_ic.html">modtest_ic</a></span>(<span class="dt">biotmle =</span> rnaseqTMLEout)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="../reference/toptable.html">toptable</a></span>(limmaTMLEout))</a></code></pre></div>
<pre><code>## # A tibble: 6 x 7
##   logFC AveExpr     t P.Value adj.P.Val     B ID         
##   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;      
## 1 -57.8   -57.8 -3.29 0.00121     0.930 -4.59 result.673 
## 2 -55.9   -55.9 -3.20 0.00160     0.930 -4.60 result.1091
## 3  62.4    62.4  3.19 0.00168     0.930 -4.60 result.482 
## 4 -59.7   -59.7 -3.10 0.00221     0.930 -4.60 result.1774
## 5 -57.6   -57.6 -3.06 0.00255     0.930 -4.60 result.991 
## 6 -77.2   -77.2 -2.98 0.00327     0.930 -4.60 result.1109</code></pre>
<hr>
</div>
<div id="results-data-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#results-data-visualization" class="anchor"></a>Results: Data Visualization</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_adj"</span>)</a></code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/pval_hist_limma_adjp-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_raw"</span>)</a></code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/pval_hist_limma_rawp-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">varInt_index &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(se)) <span class="op">%in%</span><span class="st"> "exp_var"</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">designVar &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(se))[, varInt_index]</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">design &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(designVar <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">max</a></span>(designVar))</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="kw"><a href="../reference/heatmap_ic.html">heatmap_ic</a></span>(<span class="dt">x =</span> limmaTMLEout,</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">           <span class="dt">row.dendrogram =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">           <span class="dt">clustering.method =</span> <span class="st">"hierarchical"</span>,</a>
<a class="sourceLine" id="cb12-8" data-line-number="8">           <span class="dt">design =</span> design, <span class="dt">FDRcutoff =</span> <span class="fl">1.0</span>, <span class="dt">top =</span> <span class="dv">10</span>)</a></code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/heatmap_limma_results-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/volcano_ic.html">volcano_ic</a></span>(<span class="dt">biotmle =</span> limmaTMLEout)</a></code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/volcano_plot_limma_results-1.png" width="700"></p>
<hr>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<pre><code>## R version 3.5.3 (2019-03-11)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.2 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
##  [3] BiocParallel_1.16.5         matrixStats_0.54.0         
##  [5] Biobase_2.42.0              GenomicRanges_1.34.0       
##  [7] GenomeInfoDb_1.18.1         IRanges_2.16.0             
##  [9] S4Vectors_0.20.1            BiocGenerics_0.28.0        
## [11] biotmleData_1.6.0           biotmle_1.7.2              
## [13] dplyr_0.8.0.1               BiocStyle_2.10.0           
## 
## loaded via a namespace (and not attached):
##  [1] ggdendro_0.1-20        foreach_1.4.4          assertthat_0.2.1      
##  [4] BiocManager_1.30.4     GenomeInfoDbData_1.2.0 superheat_0.1.0       
##  [7] yaml_2.2.0             globals_0.12.4         pillar_1.3.1          
## [10] backports_1.1.3        lattice_0.20-38        glue_1.3.1            
## [13] limma_3.38.3           digest_0.6.18          XVector_0.22.0        
## [16] colorspace_1.4-1       htmltools_0.3.6        Matrix_1.2-17         
## [19] plyr_1.8.4             pkgconfig_2.0.2        listenv_0.7.0         
## [22] bookdown_0.9.1         zlibbioc_1.28.0        purrr_0.3.2           
## [25] scales_1.0.0           tibble_2.1.1           ggplot2_3.1.0         
## [28] lazyeval_0.2.2         cli_1.1.0              magrittr_1.5          
## [31] crayon_1.3.4           memoise_1.1.0.9000     evaluate_0.13         
## [34] fs_1.2.7               future_1.12.0          fansi_0.4.0           
## [37] MASS_7.3-51.3          SuperLearner_2.0-24    xml2_1.2.0            
## [40] tools_3.5.3            stringr_1.4.0          munsell_0.5.0         
## [43] ggsci_2.9              compiler_3.5.3         pkgdown_1.3.0         
## [46] rlang_0.3.3            grid_3.5.3             RCurl_1.95-4.12       
## [49] iterators_1.0.10       labeling_0.3           bitops_1.0-6          
## [52] rmarkdown_1.12         gtable_0.3.0           codetools_0.2-16      
## [55] roxygen2_6.1.1         R6_2.4.0               nnls_1.4              
## [58] knitr_1.22             doFuture_0.8.0         utf8_1.1.4            
## [61] commonmark_1.7         rprojroot_1.3-2        desc_1.2.0            
## [64] stringi_1.4.3          tmle_1.3.0-2           Rcpp_1.0.1            
## [67] tidyselect_0.2.5       xfun_0.6</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#simulation-data-structure">Simulation: Data Structure</a></li>
      <li><a href="#processing-data-transformation">Processing: Data Transformation</a></li>
      <li><a href="#analysis-assessing-the-effect-of-exposure">Analysis: Assessing the Effect of Exposure</a></li>
      <li><a href="#results-data-visualization">Results: Data Visualization</a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
