<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing and Analyzing RNA-Seq Data â€¢ biotmle</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">biotmle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable</a>
    </li>
    <li>
      <a href="../articles/rnaseqProcessing.html">Processing and Analyzing RNA-Seq Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nhejazi/biotmle">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Processing and Analyzing RNA-Seq Data</h1>
                        <h4 class="author">Nima Hejazi &amp; Alan Hubbard</h4>
            
            <h4 class="date">2017-07-28</h4>
          </div>

    
    
<div class="contents">
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre><code>## biotmle: Moderated Statistics and Targeted Learning for Biomarker Discovery</code></pre>
<pre><code>## Version: 1.1.2</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: 'IRanges'</code></pre>
<pre><code>## The following objects are masked from 'package:dplyr':
## 
##     collapse, desc, slice</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## The following object is masked from 'package:dplyr':
## 
##     count</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     apply</code></pre>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Here, we briefly work through how to use the <code>biotmle</code> package with data generated by next-generation sequencing technologies, which, in contrast to microarray technologies, produce measurements in the form of discrete counts.</p>
<hr>
</div>
<div id="simulation-data-structure" class="section level2">
<h2 class="hasAnchor">
<a href="#simulation-data-structure" class="anchor"></a>Simulation: Data Structure</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">6423709</span>)
n &lt;-<span class="st"> </span><span class="dv">50</span>
g &lt;-<span class="st"> </span><span class="dv">2500</span>
cases_pois &lt;-<span class="st"> </span><span class="dv">50</span>
controls_pois &lt;-<span class="st"> </span><span class="dv">10</span>

ngs_cases &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">matrix</span>(<span class="kw">replicate</span>(n, <span class="kw">rpois</span>(g, cases_pois)), g))
ngs_controls &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">matrix</span>(<span class="kw">replicate</span>(n, <span class="kw">rpois</span>(g, controls_pois)), g))

ngs_data &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(ngs_cases, ngs_controls))
exp_var &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>, n), <span class="kw">rep</span>(<span class="dv">0</span>, n))
batch &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, n)
covar &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, n <span class="op">*</span><span class="st"> </span><span class="dv">2</span>)
design &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(exp_var, batch, covar))

<span class="kw">head</span>(ngs_data[, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>])</code></pre></div>
<pre><code>##   V1 V2 V3 V4 V5 V6 V7
## 1 69 61 39 46 50 57 43
## 2 42 59 50 54 44 53 54
## 3 41 49 52 54 50 58 34
## 4 30 44 49 44 46 36 61
## 5 50 56 45 44 46 61 58
## 6 61 50 47 53 49 54 67</code></pre>
<hr>
</div>
<div id="processing-data-transformation" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-data-transformation" class="anchor"></a>Processing: Data Transformation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(<span class="dt">assays =</span> <span class="kw">list</span>(<span class="dt">counts =</span> <span class="kw">DataFrame</span>(ngs_data)),
                           <span class="dt">colData =</span> <span class="kw">DataFrame</span>(design))
se</code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 2500 100 
## metadata(0):
## assays(1): counts
## rownames: NULL
## rowData names(0):
## colnames(100): V1 V2 ... V49.1 V50.1
## colData names(3): exp_var batch covar</code></pre>
<hr>
</div>
<div id="analysis-assessing-the-effect-of-exposure" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-assessing-the-effect-of-exposure" class="anchor"></a>Analysis: Assessing the Effect of Exposure</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rnaseqTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/biomarkertmle.html">biomarkertmle</a></span>(<span class="dt">se =</span> se,
                               <span class="dt">varInt =</span> <span class="dv">1</span>,
                               <span class="dt">type =</span> <span class="st">"exposure"</span>,
                               <span class="dt">ngscounts =</span> <span class="ot">TRUE</span>,
                               <span class="dt">parallel =</span> <span class="ot">TRUE</span>,
                               <span class="dt">family =</span> <span class="st">"gaussian"</span>,
                               <span class="dt">g_lib =</span> <span class="kw">c</span>(<span class="st">"SL.mean"</span>, <span class="st">"SL.glm"</span>, <span class="st">"SL.randomForest"</span>),
                               <span class="dt">Q_lib =</span> <span class="kw">c</span>(<span class="st">"SL.mean"</span>, <span class="st">"SL.glm"</span>, <span class="st">"SL.randomForest"</span>,
                                         <span class="st">"SL.nnet"</span>)
                              )
<span class="kw">head</span>(rnaseqTMLEout<span class="op">@</span>tmleOut<span class="op">$</span>E[, <span class="kw">seq_len</span>(<span class="dv">6</span>)])</code></pre></div>
<pre><code>##                [,1]       [,2]      [,3]       [,4]      [,5]       [,6]
## result.1 -329.83655 -212.28070 160.39144   32.96832 -36.05786 -146.77124
## result.2   91.43805 -129.00196 -39.20680  -47.18052  44.29473  -30.83862
## result.3  102.01072 -294.65959 -77.65629 -376.36991 -61.66602 -441.74551
## result.4  323.94443   93.76761  13.80324   93.71017  47.56326  224.33603
## result.5   11.71456 -148.33216  93.43461   47.79556  61.82086 -229.99698
## result.6 -183.73280   -4.00971  45.02578  -53.02987  -4.00971  -69.37162</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">limmaTMLEout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/modtest_ic.html">modtest_ic</a></span>(<span class="dt">biotmle =</span> rnaseqTMLEout)
<span class="kw">head</span>(limmaTMLEout<span class="op">@</span>topTable)</code></pre></div>
<pre><code>##               logFC    AveExpr          t    P.Value adj.P.Val         B
## result.1 -32.072167 -32.072167 -1.7430776 0.08297431 0.9303074 -4.595093
## result.2 -25.903359 -25.903359 -1.4289259 0.15470453 0.9342061 -4.595106
## result.3 -15.865179 -15.865179 -0.6675294 0.50526227 0.9731485 -4.595127
## result.4  20.394523  20.394523  0.9768991 0.32988929 0.9401033 -4.595120
## result.5 -23.932669 -23.932669 -1.2276717 0.22112330 0.9401033 -4.595113
## result.6  -8.540187  -8.540187 -0.4839883 0.62896415 0.9746949 -4.595130
##               IDs
## result.1 result.1
## result.2 result.2
## result.3 result.3
## result.4 result.4
## result.5 result.5
## result.6 result.6</code></pre>
<hr>
</div>
<div id="results-data-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#results-data-visualization" class="anchor"></a>Results: Data Visualization</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_adj"</span>)</code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/pval_hist_limma_adjp-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">type =</span> <span class="st">"pvals_raw"</span>)</code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/pval_hist_limma_rawp-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varInt_index &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">names</span>(<span class="kw">colData</span>(se)) <span class="op">%in%</span><span class="st"> "exp_var"</span>)
designVar &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(se))[, varInt_index]
design &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(designVar <span class="op">==</span><span class="st"> </span><span class="kw">max</span>(designVar))

<span class="kw"><a href="../reference/heatmap_ic.html">heatmap_ic</a></span>(<span class="dt">x =</span> limmaTMLEout, <span class="dt">design =</span> design, <span class="dt">FDRcutoff =</span> <span class="fl">1.0</span>, <span class="dt">top =</span> <span class="dv">10</span>)</code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/heatmap_limma_results-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/volcano_ic.html">volcano_ic</a></span>(<span class="dt">biotmle =</span> limmaTMLEout)</code></pre></div>
<p><img src="rnaseqProcessing_files/figure-html/volcano_plot_limma_results-1.png" width="672"></p>
<hr>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<pre><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-apple-darwin16.6.0 (64-bit)
## Running under: macOS Sierra 10.12.5
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2               SummarizedExperiment_1.6.3
##  [3] DelayedArray_0.2.7         matrixStats_0.52.2        
##  [5] Biobase_2.36.2             GenomicRanges_1.28.4      
##  [7] GenomeInfoDb_1.12.2        IRanges_2.10.2            
##  [9] S4Vectors_0.14.3           BiocGenerics_0.22.0       
## [11] biotmleData_0.99.1         biotmle_1.1.2             
## [13] dplyr_0.7.2               
## 
## loaded via a namespace (and not attached):
##  [1] superheat_0.1.0          lattice_0.20-35         
##  [3] colorspace_1.3-2         htmltools_0.3.6         
##  [5] yaml_2.1.14              rlang_0.1.1.9000        
##  [7] glue_1.1.1               DBI_0.7                 
##  [9] GenomeInfoDbData_0.99.0  foreach_1.4.3           
## [11] bindr_0.1                plyr_1.8.4              
## [13] stringr_1.2.0.9000       zlibbioc_1.22.0         
## [15] munsell_0.4.3            gtable_0.2.0            
## [17] tmle_1.2.0-5             codetools_0.2-15        
## [19] evaluate_0.10.1          labeling_0.3            
## [21] knitr_1.16               doParallel_1.0.10       
## [23] Rcpp_0.12.12             backports_1.1.0         
## [25] scales_0.4.1.9002        limma_3.32.4            
## [27] XVector_0.16.0           ggplot2_2.2.1           
## [29] digest_0.6.12            stringi_1.1.5           
## [31] grid_3.4.1               rprojroot_1.2           
## [33] tools_3.4.1              bitops_1.0-6            
## [35] magrittr_1.5             RCurl_1.95-4.8          
## [37] lazyeval_0.2.0           tibble_1.3.3.9001       
## [39] ggdendro_0.1-20          wesanderson_0.3.4       
## [41] pkgconfig_2.0.1          MASS_7.3-47             
## [43] Matrix_1.2-10            data.table_1.10.4       
## [45] SuperLearner_2.0-23-9000 nnls_1.4                
## [47] assertthat_0.2.0         rmarkdown_1.6           
## [49] iterators_1.0.8          R6_2.2.2                
## [51] compiler_3.4.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#simulation-data-structure">Simulation: Data Structure</a></li>
      <li><a href="#processing-data-transformation">Processing: Data Transformation</a></li>
      <li><a href="#analysis-assessing-the-effect-of-exposure">Analysis: Assessing the Effect of Exposure</a></li>
      <li><a href="#results-data-visualization">Results: Data Visualization</a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
