<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Biomarker Evaluation with Targeted Minimum Loss Estimation of the ATE — biomarkertmle • biotmle</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Biomarker Evaluation with Targeted Minimum Loss Estimation of the ATE — biomarkertmle" />
<meta property="og:description" content="Computes the causal target parameter defined as the difference between the
biomarker expression values under treatment and those same values under no
treatment, using Targeted Minimum Loss Estimation." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biotmle</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable with `biotmle`</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Biomarker Evaluation with Targeted Minimum Loss Estimation of the ATE</h1>
    
    <div class="hidden name"><code>biomarkertmle.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Computes the causal target parameter defined as the difference between the
biomarker expression values under treatment and those same values under no
treatment, using Targeted Minimum Loss Estimation.</p>
    </div>

    <pre class="usage"><span class='fu'>biomarkertmle</span>(
  <span class='no'>se</span>,
  <span class='no'>varInt</span>,
  <span class='kw'>normalized</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>ngscounts</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>parallel</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>bppar_type</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>future_param</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>bppar_debug</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>cv_folds</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>g_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SL.mean"</span>, <span class='st'>"SL.glm"</span>, <span class='st'>"SL.bayesglm"</span>, <span class='st'>"SL.glmnet"</span>),
  <span class='kw'>Q_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SL.mean"</span>, <span class='st'>"SL.bayesglm"</span>, <span class='st'>"SL.earth"</span>, <span class='st'>"SL.xgboost"</span>),
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>se</th>
      <td><p>A <code>SummarizedExperiment</code> containing microarray expression
or next-generation sequencing data in the <code>assays</code> slot and a matrix
of phenotype-level data in the <code>colData</code> slot.</p></td>
    </tr>
    <tr>
      <th>varInt</th>
      <td><p>A <code>numeric</code> indicating the column of the design matrix
corresponding to the treatment or outcome of interest (in the
<code>colData</code> slot of the <code>SummarizedExperiment</code> argument "se").</p></td>
    </tr>
    <tr>
      <th>normalized</th>
      <td><p>A <code>logical</code> indicating whether the data included in
the <code>assay</code> slot of the input <code>SummarizedExperiment</code> object has
been normalized externally. The default is set to <code>TRUE</code> with the
expectation that an appropriate normalization method has been applied. If
set to <code>FALSE</code>, median normalization is performed for microarray data.</p></td>
    </tr>
    <tr>
      <th>ngscounts</th>
      <td><p>A <code>logical</code> indicating whether the data are counts
generated from a next-generation sequencing experiment (e.g., RNA-seq). The
default setting assumes continuous expression measures as generated by
microarray platforms.</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>A <code>logical</code> of whether or not to use parallelization in
the estimation procedure. Invoking parallelization happens through a
combination of calls to <span class="pkg">future</span> and <span class="pkg">BiocParallel</span>. If this
argument is set to <code>TRUE</code>, <code><a href='https://rdrr.io/pkg/future/man/multiprocess.html'>multiprocess</a></code> is used,
and, if <code>FALSE</code>, <code><a href='https://rdrr.io/pkg/future/man/sequential.html'>sequential</a></code> is used, alongside
<code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code>. Other options for evaluation through
futures may be invoked by setting the argument <code>future_param</code>.</p></td>
    </tr>
    <tr>
      <th>bppar_type</th>
      <td><p>A <code>character</code> specifying the type of parallelization
backend to be invoked by <code>BiocParallel</code>. Consult the manual page for
<code><a href='https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html'>BiocParallelParam</a></code> for possible types and
descriptions on their appropriate uses. The default for this argument is
<code>NULL</code>, which silently uses <code><a href='https://rdrr.io/pkg/BiocParallel/man/DoparParam-class.html'>DoparParam</a></code>.</p></td>
    </tr>
    <tr>
      <th>future_param</th>
      <td><p>A <code>character</code> specifying the parallelization type
to be invoked when using futures for evaluation. For a list of available
types, please consult the documentation for <code><a href='https://rdrr.io/pkg/future/man/plan.html'>plan</a></code>. The
default setting (this argument set to <code>NULL</code>) silently invokes
<code><a href='https://rdrr.io/pkg/future/man/multiprocess.html'>multiprocess</a></code>.</p></td>
    </tr>
    <tr>
      <th>bppar_debug</th>
      <td><p>A <code>logical</code> indicating whether or not to rely upon
pkgBiocParallel. Setting this argument to <code>TRUE</code>, replaces the call
to <code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code> by a call to <code>lapply</code>, which
significantly reduces the overhead of debugging. Note that invoking this
option overrides all other parallelization arguments.</p></td>
    </tr>
    <tr>
      <th>cv_folds</th>
      <td><p>A <code>numeric</code> scalar indicating how many folds to use in
performing targeted minimum loss estimation. Cross-validated estimates have
been demonstrated to allow relaxation of certain theoretical conditions and
and accommodate the construction of more conservative variance estimates.</p></td>
    </tr>
    <tr>
      <th>g_lib</th>
      <td><p>A <code>character</code> vector specifying the library of machine
learning algorithms for use in fitting the propensity score P(A = a | W).</p></td>
    </tr>
    <tr>
      <th>Q_lib</th>
      <td><p>A <code>character</code> vector specifying the library of machine
learning algorithms for use in fitting the outcome regression E[Y | A,W].</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments to be passed to <code><a href='https://rdrr.io/pkg/drtmle/man/drtmle.html'>drtmle</a></code>
in computing the targeted minimum loss estimator of the average treatment
effect.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>S4 object of class <code>biotmle</code>, inheriting from
 <code>SummarizedExperiment</code>, with additional slots <code>tmleOut</code> and
 <code>call</code>, among others, containing TML estimates of the ATE of exposure
 on biomarker expression.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>dplyr</span>)</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘dplyr’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    count</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:GenomicRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:GenomeInfoDb’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:IRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    collapse, desc, intersect, setdiff, slice, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:S4Vectors’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    first, intersect, rename, setdiff, setequal, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:BiocGenerics’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine, intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    filter, lag</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, setequal, union</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>biotmleData</span>)
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>SuperLearner</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: nnls</span></div><div class='output co'>#&gt; <span class='message'>Super Learner</span></div><div class='output co'>#&gt; <span class='message'>Version: 2.0-26</span></div><div class='output co'>#&gt; <span class='message'>Package created on 2019-10-27</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>SummarizedExperiment</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>illuminaData</span>)

<span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span>(<span class='no'>illuminaData</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span>(<span class='no'>illuminaData</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>() <span class='kw'>%&gt;%</span>
  <span class='kw pkg'>dplyr</span><span class='kw ns'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span>(<span class='kw'>age</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/numeric.html'>as.numeric</a></span>(<span class='no'>age</span> <span class='kw'>&gt;</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span>(<span class='no'>age</span>))) <span class='kw'>%&gt;%</span>
  <span class='fu'>DataFrame</span>()
<span class='no'>benz_idx</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span>(<span class='no'>illuminaData</span>)) <span class='kw'>%in%</span> <span class='st'>"benzene"</span>)

<span class='no'>biomarkerTMLEout</span> <span class='kw'>&lt;-</span> <span class='fu'>biomarkertmle</span>(
  <span class='kw'>se</span> <span class='kw'>=</span> <span class='no'>illuminaData</span>[<span class='fl'>1</span>:<span class='fl'>2</span>, ],
  <span class='kw'>varInt</span> <span class='kw'>=</span> <span class='no'>benz_idx</span>,
  <span class='kw'>parallel</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>g_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SL.mean"</span>, <span class='st'>"SL.glm"</span>),
  <span class='kw'>Q_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SL.bayesglm"</span>, <span class='st'>"SL.glmnet"</span>)
)</div><div class='output co'>#&gt; <span class='warning'>Warning: Sequential evaluation over many probes may take a long time.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: globals</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: future</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘future’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:SummarizedExperiment’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:GenomicRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:IRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:S4Vectors’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: foreach</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: iterators</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: nloptr</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: quadprog</span></div><div class='output co'>#&gt; <span class='message'>Loading required namespace: arm</span></div><div class='output co'>#&gt; <span class='message'>Loading required namespace: glmnet</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard, Mark van der Laan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


