<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Biomarker Evaluation with Targeted Minimum Loss-Based Estimation of the ATE — biomarkertmle • biotmle</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Biomarker Evaluation with Targeted Minimum Loss-Based Estimation of the ATE — biomarkertmle" />

<meta property="og:description" content="Computes the causal target parameter defined as the difference between the
biomarker expression values under treatment and those same values under no
treatment, using Targeted Minimum Loss-Based Estimation." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biotmle</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/exposureBiomarkers.html">Identifying Biomarkers from an Exposure Variable</a>
    </li>
    <li>
      <a href="../articles/rnaseqProcessing.html">Processing and Analyzing RNA-Seq Data</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Biomarker Evaluation with Targeted Minimum Loss-Based Estimation of the ATE</h1>
    
    <div class="hidden name"><code>biomarkertmle.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Computes the causal target parameter defined as the difference between the
biomarker expression values under treatment and those same values under no
treatment, using Targeted Minimum Loss-Based Estimation.</p>
    
    </div>

    <pre class="usage"><span class='fu'>biomarkertmle</span>(<span class='no'>se</span>, <span class='no'>varInt</span>, <span class='kw'>normalized</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>ngscounts</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>parallel</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>bppar_type</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>future_param</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>family</span> <span class='kw'>=</span> <span class='st'>"gaussian"</span>, <span class='kw'>subj_ids</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>g_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"SL.mean"</span>,
  <span class='st'>"SL.glm"</span>, <span class='st'>"SL.earth"</span>), <span class='kw'>Q_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"SL.mean"</span>, <span class='st'>"SL.glm"</span>, <span class='st'>"SL.earth"</span>,
  <span class='st'>"SL.ranger"</span>), <span class='kw'>ci_type</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"normal"</span>, <span class='st'>"logistic"</span>, <span class='st'>"bernstein"</span>, <span class='st'>"bennett"</span>,
  <span class='st'>"hoeffding"</span>), <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>se</th>
      <td><p>(SummarizedExperiment) - containing expression or next-generation
sequencing data in the "assays" slot and a matrix of phenotype-level data
in the "colData" slot.</p></td>
    </tr>
    <tr>
      <th>varInt</th>
      <td><p>(numeric) - indicating the column of the design matrix
corresponding to the treatment or outcome of interest (in the
<code>colData</code> slot of the <code>SummarizedExperiment</code> argument "se").</p></td>
    </tr>
    <tr>
      <th>normalized</th>
      <td><p>(logical) - whether the data included in the <code>assay</code>
slot of the input <code>SummarizedExperiment</code> object has been normalized
already. The default is set to <code>TRUE</code> since it is expected that most
practitioners would apply normalization methods appropriate to the type of
assay being analyzed. If set to <code>FALSE</code>, median normalization is
performed for microarray (i.e., non-RNA-seq) data.</p></td>
    </tr>
    <tr>
      <th>ngscounts</th>
      <td><p>(logical) - whether the data are counts generated from a
next-generation sequencing (NGS) experiment (e.g., RNA-seq). The default
setting assumes continuous expression measures as generated by platforms
that are microarray-type (i.e., so-called "targeted" assays).</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>(logical) - whether or not to use parallelization in the
estimation procedure. Invoking parallelization happens through a
combination of calls to <code>future</code> and <code>BiocParallel</code>. If this
argument is set to <code>TRUE</code>, <code><a href='https://www.rdocumentation.org/packages/future/topics/multiprocess'>future::multiprocess</a></code> is used,
and if <code>FALSE</code>, <code><a href='https://www.rdocumentation.org/packages/future/topics/sequential'>future::sequential</a></code> is used, alongside
<code><a href='https://www.rdocumentation.org/packages/BiocParallel/topics/bplapply'>BiocParallel::bplapply</a></code>. Other options for evaluation through
futures may be invoked by setting the argument <code>future_param</code>.</p></td>
    </tr>
    <tr>
      <th>bppar_type</th>
      <td><p>(character) - specifies the type of backend to be used with
the parallelization invoked by <code>BiocParallel</code>. Consult the manual page
for <code><a href='https://www.rdocumentation.org/packages/BiocParallel/topics/BiocParallelParam-class'>BiocParallel::BiocParallelParam</a></code> for possible types and
descriptions on their appropriate uses. The default for this argument is
<code>NULL</code>, which silently uses <code><a href='https://www.rdocumentation.org/packages/BiocParallel/topics/DoparParam-class'>BiocParallel::DoparParam</a></code>.</p></td>
    </tr>
    <tr>
      <th>future_param</th>
      <td><p>(character) - specifies the type of parallelization to be
invoked when using futures for evaluation. For a list of the available
types, please consult the documentation for <code><a href='https://www.rdocumentation.org/packages/future/topics/plan'>future::plan</a></code>. The default
setting (this argument set to <code>NULL</code>) silently invokes
<code><a href='https://www.rdocumentation.org/packages/future/topics/multiprocess'>future::multiprocess</a></code>. Be careful if changing this setting.</p></td>
    </tr>
    <tr>
      <th>family</th>
      <td><p>(character) - specification of error family: "binomial" or
"gaussian".</p></td>
    </tr>
    <tr>
      <th>subj_ids</th>
      <td><p>(numeric vector) - subject IDs to be passed directly to
subject should have the exact same numerical identifier; coerced to class
<code>numeric</code> if not provided in the appropriate form.</p></td>
    </tr>
    <tr>
      <th>g_lib</th>
      <td><p>(char vector) - library of learning algorithms to be used in
fitting the propensity score E[A | W] (the nuisance parameter denoted "g" in
the literature on targeted minimum loss-based estimation).</p></td>
    </tr>
    <tr>
      <th>Q_lib</th>
      <td><p>(char vector) - library of learning algorithms to be used in
fitting the outcome regression E[Y | A, W] (the nuisance parameter denoted
"Q" in the literature on targeted minimum loss-based estimation).</p></td>
    </tr>
    <tr>
      <th>ci_type</th>
      <td><p>The type of confidence interval to be constructed. By default,
confidence intervals are generated based on the normal approximation but
more conservative inference is attainable (and preferable under deviations
from normality) based on concentration inequalities (e.g., the procedures of
Hoeffding, Bernstein, or Bennett) or alternative reference distributions
(e.g., logistic).</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments to be passed directly to <code><a href='https://www.rdocumentation.org/packages/tmle/topics/tmle'>tmle::tmle</a></code> in
fitting the targeted minimum loss-based estimator of the average treatment
effect. Consult the documentation of that function for details.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>S4 object of class <code>biotmle</code>, generated by sub-classing
 <code>SummarizedExperiment</code>, with additional slots containing <code>tmleOut</code>
 and <code>call</code>, among others, containing TMLE-based estimates of the
 relationship between a biomarker and exposure or outcome variable and the
 original call to this function (for user reference), respectively.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>dplyr</span>)</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘dplyr’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    count</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:GenomicRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:GenomeInfoDb’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:IRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    collapse, desc, intersect, setdiff, slice, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:S4Vectors’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    first, intersect, rename, setdiff, setequal, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:BiocGenerics’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine, intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:testthat’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    matches</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    filter, lag</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, setequal, union</span></div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>biotmleData</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>illuminaData</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>SummarizedExperiment</span>)
<span class='st'>"%ni%"</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/funprog'>Negate</a></span>(<span class='st'>"%in%"</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>illuminaData</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>illuminaData</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/data.frame'>data.frame</a></span>() <span class='kw'>%&gt;%</span>
  <span class='kw pkg'>dplyr</span><span class='kw ns'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span>(<span class='kw'>age</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/numeric'>as.numeric</a></span>(<span class='no'>age</span> <span class='kw'>&gt;</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/median'>median</a></span>(<span class='no'>age</span>))) <span class='kw'>%&gt;%</span>
  <span class='fu'>DataFrame</span>()

<span class='no'>varInt_index</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/which'>which</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>illuminaData</span>)) <span class='kw'>%in%</span> <span class='st'>"benzene"</span>)

<span class='no'>biomarkerTMLEout</span> <span class='kw'>&lt;-</span> <span class='fu'>biomarkertmle</span>(
  <span class='kw'>se</span> <span class='kw'>=</span> <span class='no'>illuminaData</span>[<span class='fl'>1</span>:<span class='fl'>2</span>, ],
  <span class='kw'>varInt</span> <span class='kw'>=</span> <span class='no'>varInt_index</span>,
  <span class='kw'>parallel</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>family</span> <span class='kw'>=</span> <span class='st'>"gaussian"</span>,
  <span class='kw'>g_lib</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"SL.mean"</span>, <span class='st'>"SL.glm"</span>),
  <span class='kw'>Q_lib</span> <span class='kw'>=</span> <span class='st'>"SL.glm"</span>
)</div><div class='output co'>#&gt; <span class='warning'>Warning: Sequential evaluation is strongly discouraged. </span>
#&gt; <span class='warning'> Proceed with caution.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: globals</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: future</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘future’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:SummarizedExperiment’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:GenomicRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:IRanges’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:S4Vectors’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    values</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: foreach</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: iterators</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: nnls</span></div><div class='input'>#
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nima Hejazi, Alan Hubbard.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

