utils::globalVariables(c("gene","assay<-"))

#' Biomarker Evaluation with Targeted Minimum Loss-Based Estimation (TMLE)
#'
#' Computes the causal target parameter defined as the difference between the
#' biomarker expression values under treatment and those same values under no
#' treatment, using Targeted Minimum Loss-Based Estimation.
#'
#' @param se (SummarizedExperiment) - containing expression or next-generation
#'        sequencing data in the "assays" slot and a matrix of phenotype-level
#'        data in the "colData" slot.
#' @param varInt (numeric) - indicating the column of the design matrix
#'        corresponding to the treatment or outcome of interest (in the
#'        "colData" slot of the "se" argument above).
#' @param type (character) - choice of the type of TMLE to perform: "exposure"
#'        to identify biomarkers related to an exposure (input as A), or
#'        "outcome" to identify biomarkers related to an outcome (input as Y).
#' @param ngscounts (logical) - whether the data are counts generated from a
#'        next-generation sequencing (NGS) experiment (e.g., RNA-seq). The
#'        default setting assumes continuous expression measures as generated by
#'        microarray-type platforms.
#' @param parallel (logical, numeric) - whether to use or the number of cores to
#'        be used when the TMLE-based estimation procedure is parallelized.
#' @param family (character) - specification of error family: "binomial" or
#'        "gaussian".
#' @param g_lib (char vector) - library of learning algorithms to be used in
#'        fitting the "g" step of the standard TMLE procedure.
#' @param Q_lib (char vector) - library of learning algorithms to be used in
#'        fitting the "Q" step of the standard TMLE procedure.
#'
#' @importFrom parallel detectCores
#' @importFrom doParallel registerDoParallel
#' @importFrom foreach foreach "%dopar%"
#' @importFrom SummarizedExperiment assay colData rowData SummarizedExperiment
#'
#' @return S4 object of class \code{biotmle}, generated by sub-classing
#'         \code{SummarizedExperiment}, with additional slots containing
#'         \code{tmleOut} and \code{call}, among others, containing TMLE-based
#'         estimates of the relationship between a biomarker and exposure or
#'         outcome variable and the original call to this function (for user
#'         reference), respectively.
#'
#' @export biomarkertmle
#'
#' @examples
#' library(dplyr)
#' library(biotmleData)
#' data(illuminaData)
#' library(SummarizedExperiment)
#' "%ni%" = Negate("%in%")
#'
#' colData(illuminaData) <- colData(illuminaData) %>%
#'      data.frame %>%
#'      dplyr::mutate(age = as.numeric(age > median(age))) %>%
#'      DataFrame
#'
#' varInt_index <- which(names(colData(illuminaData)) %in% "benzene")
#'
#' biomarkerTMLEout <- biomarkertmle(se = illuminaData[1, ],
#'                                   varInt = varInt_index,
#'                                   type = "exposure",
#'                                   parallel = 1,
#'                                   family = "gaussian",
#'                                   g_lib = c("SL.mean"),
#'                                   Q_lib = c("SL.mean")
#'                                  )
#'
biomarkertmle <- function(se,
                          varInt,
                          type = c("exposure", "outcome"),
                          ngscounts = FALSE,
                          parallel = TRUE,
                          family = "gaussian",
                          g_lib = c("SL.glm", "SL.randomForest", "SL.nnet",
                                    "SL.polymars", "SL.mean"),
                          Q_lib = c("SL.glm", "SL.randomForest", "SL.nnet",
                                    "SL.mean")
                          ) {

  # ============================================================================
  # catch input and return in output object for user convenience
  # ============================================================================
  call <- match.call(expand.dots = TRUE)

  # ============================================================================
  # explicitly check that the "type" argument is of the appropriate type
  # ============================================================================
  type <- match.arg(type)

  # ============================================================================
  # invoke S4 class constructor for "bioTMLE" object
  # ============================================================================
  biotmle <- .biotmle(
       SummarizedExperiment(
          assays = list(expMeasures = assay(se)),
          rowData = rowData(se),
          colData = colData(se)
       ),
       call = call,
       tmleOut = as.data.frame(matrix(NA, 10, 10)),
       topTable = as.data.frame(matrix(NA, 10, 10))
   )

  # ============================================================================
  # invoke the voom transform from LIMMA if next-generation sequencing data)
  # ============================================================================
  if (ngscounts) {
    voom_out <- rnaseq_ic(biotmle)
    voom_exp <- 2^(voom_out$E)
    assay(se) <- voom_exp
  }

  #=============================================================================
  # set up parallelization based on input
  # ============================================================================
  if (class(parallel) == "numeric") doParallel::registerDoParallel(parallel)
  if (class(parallel) == "logical") {
    nCores <- parallel::detectCores()
    if (nCores > 1) {
      doParallel::registerDoParallel(nCores)
    } else {
      warning("option 'parallel' is set to TRUE but only 1 core detected.")
    }
    if (parallel == FALSE) {
      stop("parallelization set to FALSE: the estimation procedure will not
           run to completion in any sort of timely fashion.")
    }
  }
  #=============================================================================
  # TMLE procedure to identify biomarkers based on an EXPOSURE
  # ============================================================================
  if (type == "exposure") {

    # median normalization
    if (!ngscounts) {
      Y <- as.data.frame(t(limma::normalizeBetweenArrays(assay(se),
                                                         method = "scale")))
    } else {
      Y <- as.data.frame(t(assay(se)))
    }
    # simple sanity check of whether Y includes array values
    if(unique(lapply(Y, class)) != "numeric") {
      stop("Warning - values in Y do not appear to be numeric...")
    }

    # exposure / treatment
    A <- as.numeric(colData(se)[, varInt])

    # baseline covariates
    W <- as.data.frame(colData(se)[, -varInt])
    if(dim(W)[2] == 0) {
      W <- as.numeric(rep(1, length(A)))
    }

    # perform multi-level TMLE-based estimation for genes as Y
    biomarkerTMLEout <- foreach::foreach(gene = seq_len(ncol(Y)),
                                         .combine = cbind) %dopar% {
      print(paste("Estimating target parameter for", gene, "of", ncol(Y)))
      out <- biomarkerTMLE_exposure(Y = Y[, gene],
                                    W = W,
                                    A = A,
                                    a = unique(A),
                                    g_lib = g_lib,
                                    Q_lib = Q_lib,
                                    family = family
                                   )
    }

    if (ngscounts) {
      voom_out$E <- t(biomarkerTMLEout)
      biotmle@tmleOut <- voom_out
    } else {
      biotmle@tmleOut <- as.data.frame(t(biomarkerTMLEout))
    }
    return(biotmle)

  } else if (type == "outcome") {
  #=============================================================================
  # TMLE procedure to identify biomarkers based on an OUTCOME
  #=============================================================================

    # median normalization
    if (!ngscounts) {
      A <- as.data.frame(t(limma::normalizeBetweenArrays(assay(se),
                                                         method = "scale")))
    } else {
      A <- as.data.frame(t(assay(se)))
    }
    # simple sanity check of whether A includes array values
    if(unique(lapply(A, class)) != "numeric") {
      stop("Warning - values in A do not appear to be numeric...")
    }

    # exposure / treatment
    Y <- as.numeric(colData(se)[, varInt])

    # baseline covariates
    W <- as.data.frame(colData(se)[, -varInt])
    if(dim(W)[2] == 0) {
      W <- as.numeric(rep(1, length(Y)))
    }

    # perform multi-level TMLE-bases estimation for genes as A
    biomarkerTMLEout <- foreach::foreach(gene = seq_len(ncol(A)),
                                         .combine = rbind) %dopar% {
      print(paste("Estimating causal effect for", gene, "of", ncol(A)))
      out <- biomarkerTMLE_outcome(Y = Y,
                                   W = W,
                                   A = A[, gene],
                                   a = unique(A),
                                   g_lib = g_lib,
                                   Q_lib = Q_lib,
                                   family = family
                                  )
    }

    if (ngscounts) {
      voom_out$E <- t(biomarkerTMLEout)
      biotmle@tmleOut <- voom_out
    } else {
      biotmle@tmleOut <- as.data.frame(t(biomarkerTMLEout))
    }
    return(biotmle)
  } else {
    warning("improper input for 'type': choices are 'exposure' and 'outcome'.")
  }
}
