---
title: 'biotmle: Targeted Learning for Biomarker Discovery'
tags:
  - targeted learning
  - variable importance
  - causal inference
  - bioinformatics
  - genomics
  - R
authors:
  - name: Nima S. Hejazi
    orcid: 0000-0002-7127-2789
    affiliation: 1
  - name: Weixin Cai
    orcid: 0000-0003-2680-3066
    affiliation: 1
  - name: Alan E. Hubbard
    orcid: 0000-0002-3769-0127
    affiliation: 1
affiliations:
  - name: Division of Biostatistics, University of California, Berkeley
    index: 1
date: 08 January 2017
bibliography: paper.bib
---

```{r setup, echo=FALSE}
library(dplyr)
suppressMessages(library(SummarizedExperiment))
```

# Summary

The `biotmle` package provides an implementation of a biomarker discovery
methodology based on targeted minimum loss-Based estimation (TMLE)
[@vdl2011targeted] and a generalization of the moderated t-statistic of
[@smyth2004linear], designed for use with biological sequencing data (e.g.,
microarrays, RNA-seq). The statistical approach made available in this package
relies on the use of TMLE to rigorously evaluate the association between a set
of potential biomarkers and another variable of interest while adjusting for
potential confounding from another set of user-specified covariates. The
implementation is in the form of a package for the R language for statistical
computing [@R].

There are two principal ways in which the biomarker discovery techniques in
the `biotmle` R package can be used: to evaluate the association between (1) a
phenotypic measure (say, environmental exposure) and a biomarker of interest,
and (2) an outcome of interest (e.g., survival status at a given time) and a
biomarker measurement, both while controlling for background covariates (e.g.,
BMI, age). By using an estimation procedure based on TMLE, the package produces
results based on the Average Treatment Effect (ATE), a statistical parameter
with a well-studied causal interpretation (see @vdl2011targeted for extended
discussions), making the `biotmle` R package well-suited for applications in
bioinformatics, epidemiology, and genomics.

Let's take a look at an example data set and clean it up for use with this R
package:

```{r clean_data}
library(biotmle)
library(biotmleData)
data(illuminaData)

# discretize "age" in the phenotype-level data
colData(illuminaData) <- colData(illuminaData) %>%
  data.frame %>%
  dplyr::mutate(age = as.numeric(age > median(age))) %>%
  DataFrame

# specify column index of treatment/exposure variable of interest
varInt_index <- which(names(colData(illuminaData)) %in% "benzene")
```

We would call the principal function of this R package (`biomarkertmle`) with
the following syntax:

```{r biomarkerTMLE_eval, eval=FALSE}
biomarkerTMLEout <- biomarkertmle(se = illuminaData,
                                  varInt = varInt_index,
                                  family = "gaussian",
                                  g_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.polymars", "SL.mean"),
                                  Q_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.nnet", "SL.mean")
                                 )
```

```{r load_biomarkerTMLE_result, echo=FALSE}
data(biomarkertmleOut)
```

Note that the above call is not executed, as the estimation procedure is both
time- and resource-intensive.

We would perform a moderated test on the output of the `biomarkertmle` function
using the function `modtest_ic`:

```{r limmaTMLE_eval}
limmaTMLEout <- modtest_ic(biotmle = biomarkerTMLEout)
```

While the principal table of results produced by this R package matches those
produced by the well-known `limma` R package [@smyth2005limma], there are also
several plot methods made available for the `bioTMLE` S4 class -- subclassed
from the popular `SummarizedExperiment` class -- introduced by this package
[@huber2015orchestrating]. For illustrative purposes, we demonstrate the ouput
of two such functions on anonymized experimental data below:

```{r plot_heatmap, fig.align='center'}
varInt_index <- which(names(colData(illuminaData)) %in% "benzene")
designVar <- as.data.frame(colData(illuminaData))[, varInt_index]
designVar <- as.numeric(designVar == max(designVar))

heatmap_ic(x = limmaTMLEout, design = designVar, FDRcutoff = 0.05, top = 25)
```

The above heatmap displays the results for the top $25$ biomarkers. The package
also provides conveniences for generating volcano plots:

```{r plot_volcano, fig.align='center'}
volcano_ic(biotmle = limmaTMLEout)
```

\newpage

# References
